packages/assets-controllers/src/NftController.ts

```diff
@@ -403,30 +424,41 @@ export class NftController extends BaseController<
    *
    * @param preferencesState - The new state of the preference controller.
    * @param preferencesState.ipfsGateway - The configured IPFS gateway.
-   * @param preferencesState.openSeaEnabled - Controls whether the OpenSea API is used.
    * @param preferencesState.isIpfsGatewayEnabled - Controls whether IPFS is enabled or not.
+   * @param preferencesState.displayNftMedia - Controls whether the NFT API is used (mobile).
+   * @param preferencesState.openSeaEnabled - Controls whether the NFT API is used (extension).
    */
   async #onPreferencesControllerStateChange({
     ipfsGateway,
-    openSeaEnabled,
     isIpfsGatewayEnabled,
-  }: PreferencesState) {
+    displayNftMedia,
+    openSeaEnabled,
+  }: {
+    ipfsGateway: string;
+    isIpfsGatewayEnabled: boolean;
+    // TODO: Mobile PreferencesController uses displayNftMedia, Extension PreferencesController uses openSeaEnabled
+    // TODO: Replace this type with PreferencesState once both clients use the same PreferencesController
+    displayNftMedia?: boolean;
+    openSeaEnabled?: boolean;
```

Comment:
This is the easiest way to ensure compatibility with both extension and mobile for the time being without major refactors in extension code for PreferencesController.

It will also ensure that when those changes are done and the same state is used for both clients, things will continue to work.